### DISCUSSION TITLE
why does df -h / show way more disk space than I actually have on the drive?

### DISCUSSION THREAD

**Discussion by rbavery** [microsoft/WSL#9757] [Q&A]
I installed wsl to my I drive which I confirmed by finding the ext vhd file.

It only has 512 GB but wsl shows this

```
rave@rave:~$ df -h /
Filesystem      Size  Used Avail Use% Mounted on
/dev/sdd       1007G 1013M  955G   1% /
```

this got me into an error where I tried to download more data than my drive supported. Is something misconfigured? How can I get wsl to accurately report drive space?

**Comment by Biswa96** [comment]
WSL 2 automatically resizes these VHD files to meet storage needs. You can try to use `optimize-vhd` command in Powershell which may reduce the size of a virtual hard disk file.

**Reply by guenhter** [reply]
@Biswa96  I guess the issue is not how much space the VHD takes, but that the actual harddrive has 512GB and within WSL it shows that the disk has 955G. This seems not right.

**Reply by sgaidhan** [reply]
@Biswa96 did you get answer?? I am also seeing exactly same. its showing 955 GB but i have only 500 GB in total.

**Reply by gorsedh** [reply]
same here, shows 1007 GB, my pc only has 256...

**Reply by zcobol** [reply]
@gorsedh you are seeing the `virtual driver` size. WSL defaults to 1TB of virtual drive.

**Reply by diogo-rossi** [reply]
I'm with the exactly same issue. Did someone get the answer?

**Reply by gorsedh** [reply]
@Diogo-Rossi see the question just above yours

**Reply by diogo-rossi** [reply]
@gorsedh Thanks. But the question remains: how to get the actual size of the WSL partition?

**Reply by mattcheau** [reply]
@Diogo-Rossi Which partition? Which distro?

**Reply by diogo-rossi** [reply]
@mattcheau, I have only one default Ubuntu distro in WSL. But I just found in the internet that the disk usage of WSL in Windows drive (in this case) is the size of the file: `C:\Users\user\AppData\Local\Packages\CanonicalGroupLimited.Ubuntu_79rhkp1fndgsc\LocalState\ext4.vhdx`.
Is that right?

**Reply by mattcheau** [reply]
@Diogo-Rossi That's the best way to view the VHD size for a given installed WSL distro, yes. Official documentation on the subject can be found [here](https://learn.microsoft.com/en-us/windows/wsl/disk-space) (your specific concern addressed at the very bottom).

**Reply by sharpchen** [reply]
@Diogo-Rossi the size of vhdx might not be as large as the space subsystem actually took, it can be shrank by `diskpart`, or use `wsl --manage <distro> --set-sparse true` to let wsl release unused space automatically

**Reply by diogo-rossi** [reply]
@sharpchen . Thanks for the information! Regarding the command `wsl --manage <distro> --set-sparse true`, it seems [there are some issues with this feature](https://github.com/microsoft/WSL/issues/10609#issuecomment-2375610151)

**Reply by sharpchen** [reply]
@Diogo-Rossi Oops, didn't know that. Thanks for reminding, I will disable it.
